pkg/sql/create_type.go

```diff
@@ -99,6 +101,91 @@ func getCreateTypeParams(
 	return typeKey, id, nil
 }
 
+// When a type is created in Postgres, Postgres will implicitly create an array
+// type of that user defined type. This array type tracks changes to the
+// original type, and is dropped when the original type is dropped.
+// createArrayType creates the implicit array type for the input TypeDescriptor
+// and returns the ID of the created type.
+func (p *planner) createArrayType(
+	params runParams,
+	n *tree.CreateType,
+	typ *tree.TypeName,
+	typDesc *sqlbase.MutableTypeDescriptor,
+	db *DatabaseDescriptor,
+) (sqlbase.ID, error) {
+	// Postgres starts off trying to create the type as _<typename>. It then
+	// continues adding "_" to the front of the name until it doesn't find
+	// a collision.
+	schemaID := sqlbase.ID(keys.PublicSchemaID)
+	arrayTypeName := "_" + typ.Type()
+	var arrayTypeKey sqlbase.DescriptorKey
+	for {
```

Comment:
To be honest, this leaves a bad taste in my mouth. I think it would be better to just fail here. I think that it will be confusing for us and probably tools - I know a lot of things expect there to be a single _ in front of the type name to make an array type.

Reply:
(Fail here if there's a collision, I mean)

Reply:
are you sure? It's pretty well documented here https://www.postgresql.org/docs/9.5/sql-createtype.html.

Reply:
I'm just scared about fallout when we drop types. Maybe it's fine? It just seems like a footgun for us.

Reply:
Fair enough. I think you've thought this through much more carefully than I have.